sysvar/src/lib.rs

```diff
@@ -206,7 +217,8 @@ macro_rules! impl_sysvar_get {
 
             match result {
                 $crate::__private::SUCCESS => Ok(var),
-                e => Err(e.into()),
+                // Unexpected errors are folded into `UnsupportedSysvar`.
+                _ => Err($crate::__private::ProgramError::UnsupportedSysvar),
```

Comment:
Just to check, is this actually reachable by the syscall? Or just by the stub call? I don't see any other error code returned by the syscall implementation

Reply:
No, it should not happen. It is here only because we are matching against an `u64` value.

Reply:
And yes, it will happen in the case of the stub call since it returns `UNSUPPORTED_SYSVAR`.